pkg/cloud/libvirt/client/ignition.go

```diff
@@ -57,6 +63,199 @@ func setIgnition(domainDef *libvirtxml.Domain, client *libvirtClient, ignition *
 	return nil
 }
 
+func setIgnitionForS390X(domainDef *libvirtxml.Domain, client *libvirtClient, ignition *providerconfigv1.Ignition, kubeClient kubernetes.Interface, machineNamespace, volumeName string) error {
```

Comment:
Quite a bit of code being added for S390 here. Could we please keep it in a separate file? `ignition_s390.go` for example?

Reply:
Good point, but it doesn't work, because "*._s390x.go" or "*_s390.go" will only be complied on s390x platform, accordingly you will get a "undefined: setIgnitionForS390X" error when building the project. 

Two proposals:
1. Use a different name, i.e. "s390x_ignition.go", "ignition_s390a.go", etc..  It works, but I really don't like the names. If you insist, then I would like to use "s390x_igninition.go".
2. Keep it unchanged. 

What do you think?

Reply:
Right, good catch. `s390_ignition.go` sounds good to me.

Reply:
or better yet, name it `ignition_s390.go` and name the functions the same `setIgnition` (i-e keep the existing (in master) names), enclose the non-s390 version in a build constraint of `!s390`, and let Go compiler take care of which one to use.